tests/cpgrid/id_entity_entityrep.cpp

```diff
@@ -0,0 +1,386 @@
+//===========================================================================
+/*
+  Copyright 2025 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "config.h"
+
+#define BOOST_TEST_MODULE TemplateOverloadsForIdAndIndexTests
```

Comment:
Not critical at all, but I was wondering if "TemplateOverloadsForIdAndIndexTests" still reflects what the test does. Maybe we could pick a name that fits the current version a bit better? "EntityAndEntityRepIdComparisonTests"?